FridaTests/MacOsTargetTests.swift

```diff
@@ -0,0 +1,162 @@
+import XCTest
+@testable import Frida
+
+fileprivate enum TestScriptMessage {
+    case regular(type: String, payload: String)
+    case other(message: Any, data: Data?)
+}
+
+fileprivate class TestScriptDelegate: NSObject, ScriptDelegate {
+    var onMessage: ((TestScriptMessage) -> Void)?
+    
+    @objc func script(_ script: Script, didReceiveMessage message: Any, withData data: Data?) {
+        guard let dict = message as? [String: String] else {
+            onMessage?(.other(message: message, data: data))
+            return
+        }
+        
+        let typeMaybe = dict["type"]
+        let payloadMaybe = dict["payload"]
+        
+        guard let type = typeMaybe, let payload = payloadMaybe else {
+            print("Couldn't get type and payload")
+            return
+        }
+        
+        onMessage?(.regular(type: type, payload: payload))
+    }
+}
```

Comment:
These two classes look like details that belong below `MacOsTargetTests`. (Newspaper ordering.)